app/views/results/show.html.slim

```diff
@@ -60,20 +81,22 @@
         - accordion.with_section(heading_text: t(".capital_calculation"))
           = render "capital_table", caption_size: "m"
 
-  .govuk-button-group
-    / For eligible controlled checks there is no primary CTA in this group, as it appears below.
-    / For other checks, 'Save as PDF' is the primary CTA.
-    - if @model.level_of_help == "controlled" && @model.decision == "eligible"
-      = link_to t(".save_as_pdf"),
-                download_result_path(params[:assessment_code]),
-                target: "_blank", rel: "noopener", class: "govuk-button govuk-button--secondary"
-    - else
-      = link_to t(".save_as_pdf"), download_result_path(params[:assessment_code]), target: "_blank", rel: "noopener", class: "govuk-button"
-      = link_to t(".start_another_check"), new_check_path, class: "govuk-button govuk-button--secondary", data: { module: "govuk-button" }
+  - unless end_of_journey && !@check.immigration_or_asylum?
```

Comment:
A few issues with this. First, the previous (possibly before your time) agreement in the team has been that `unless` is harder to get your head around than `if` so we tend to only use it for simple clauses. To keep consistent, can we change `unless A && !B` to `if !A || B`.

Second, as above, I don't think `immigration_or_asylum?` is the test here, I think it's asylum support.

Third, as it stands this means if the flag is enabled, non-immigration _certificated_ checks don't get _any_ buttons on this screen. Or ineligible non-immigration controlled checks. You need the logic on line 84 to be the inverse of the logic on line 23. To make it clearer, you could maybe at the top of the page have a line saying `show_buttons_at_top = end_of_journey_flag && @check.controlled? && !@model.ineligible && !@check.asylum_support` and then on line 23 you say `if show_buttons_at_top` and on line 84 you say `if !show_buttons_at_top`

Reply:
You summary about the 'third issue' was really helpful. Thank you.